mqtt/src/main/scala/Mqtt.scala

```diff
@@ -31,6 +31,13 @@ private[mqtt] trait MqttConnectorLogic { this: GraphStageLogic =>
 
   def connectionSettings: MqttConnectionSettings
   def handleConnection(client: IMqttAsyncClient): Unit
+
+  /**
+   * Callback, that is called from the MQTT client thread before invoking
+   * message handler callback in the GraphStage context.
+   */
+  def beforeHandleMessage(): Unit
```

Comment:
Hmm. This is only to be able to do the backpressure using the semaphore so it's only for the Source, the sink will not use it. If so it's a bit weird that it is in the common connector trait.